arviz/stats/ecdf_utils.py

```diff
@@ -0,0 +1,163 @@
+"""Functions for evaluating ECDFs and their confidence bands."""
+from typing import Callable, Optional, Tuple
+import warnings
+
+import numpy as np
+from scipy.stats import uniform, binom
+
+
+def compute_ecdf(sample: np.ndarray, eval_points: np.ndarray) -> np.ndarray:
+    """Compute ECDF of the sorted `sample` at the evaluation points."""
+    return np.searchsorted(sample, eval_points, side="right") / len(sample)
+
+
+def _get_ecdf_points(
+    sample: np.ndarray, eval_points: np.ndarray, difference: bool
+) -> Tuple[np.ndarray, np.ndarray]:
+    """Compute the coordinates for the ecdf points using compute_ecdf."""
+    x = eval_points
+    y = compute_ecdf(sample, eval_points)
+
+    if not difference and y[0] > 0:
+        x = np.insert(x, 0, x[0])
+        y = np.insert(y, 0, 0)
+    return x, y
+
+
+def _simulate_ecdf(
+    ndraws: int,
+    eval_points: np.ndarray,
+    rvs: Callable[[int, Optional[np.random.RandomState]], np.ndarray],
+    random_state: Optional[np.random.RandomState] = None,
```

Comment:
Hmm, this makes the design a bit tricky. Because currently we require the `rvs` function takes a `random_state` keyword, which limits to functions that look like the ones in scipy.stats. But if we wanted to support e.g. `Generator.binom` or `RandomState.binom`, then we would require `rvs` is just a function that takes a sample size as argument and let the user decide how that is seeded. Does this latter option (i.e. not taking a `random_state` keyword at all) make more sense as a design?

Reply:
scipy `random_state` argument can be None, integers, RandomState or Generator. I don't think anything needs to change as it is passed as is to the scipy distribution

Reply:
Reference: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.rv_continuous.rvs.html#scipy.stats.rv_continuous.rvs